pkg/controllers/user/logging/generator/validator.go

```diff
@@ -124,16 +127,5 @@ func filterRubyCode(s string) error {
 }
 
 func escapeString(postDoc string) string {
-	var escapeReplacer = strings.NewReplacer(
-		"\t", `\\t`,
-		"\n", `\\n`,
-		"\r", `\\r`,
-		"\f", `\\f`,
-		"\b", `\\b`,
-		"\"", `\\\"`,
-		"\\", `\\\\`,
-	)
-
-	escapeString := escapeReplacer.Replace(postDoc)
-	return fmt.Sprintf(`"%s"`, escapeString)
+	return fmt.Sprintf("%q", postDoc)
```

Comment:
This convert is the simulation from Fluentd. Base on the doc https://docs.fluentd.org/configuration/config-file#in-double-quoted-string-literal-is-escape-character and the test case https://github.com/fluent/fluentd/blob/v1.6.3/test/config/test_literal_parser.rb

For example, original user input is `firstline\nsecondline`, and we parse to `firstline\\nsecondline`, fluentd will parse back to `firstline\nsecondline`

Reply:
https://github.com/rancher/rancher/blob/85dedb19bbdfabc5e81a8258b868cc1ca8261b0d/pkg/controllers/user/logging/generator/validator.go#L128
Take the above as an example.
Please note that the backslash inside backquote in Golang is not the escape character. so `\\t` is literally two backslashes and a `t`. When it is quoted as `"\\t"` in following codes, it becomes an escaped backslash and a `t` from the fluentd parser's view. And it does not match the original "\t" that stands for a single horizontal tab.